df.viewscreen.xml

```diff
@@ -875,19 +875,32 @@
 
     <class-type type-name='viewscreen_image_creatorst' inherits-from='viewscreen'>
         <stl-string name='title'/>
-        <stl-vector type-name='int32_t'/>
-        <int32_t name='category_idx'/>
-        <enum name='category' base-type='int32_t'>
+        <stl-vector name='modes' type-name='int32_t'/> menu items shown on screen, from the enum below
```

Comment:
Ideally this would also use the enum. I don't think that's possible with the enum defined inline, so I think what similar screens have done is just define the enum outside of the class. I'll look into that.

Reply:
See viewscreen_unitlist_page, for example